core_types/src/agent.rs

```diff
@@ -1,129 +1,64 @@
-pub struct KeyBuffer([u8; KeyBuffer::KEY_LEN]);
-
-impl KeyBuffer {
-    /// Constants specific to KeyBuffer
-    const PARITY_LEN: usize = 5;
-    const KEY_LEN: usize = 64;
-    const HALF_KEY_LEN: usize = KeyBuffer::KEY_LEN / 2;
-
-    /// take a potentially user-entered base64url encoded user representation
-    /// of an public key identity
-    /// apply reed-solomon parity correction
-    /// returns a raw byte buffer
-    pub fn with_corrected(s: &str) -> Result<KeyBuffer, HolochainError> {
-        let s = s.replace("-", "+").replace("_", "/");
-        let base64 = base64::decode(&s)?;
-        let dec = Decoder::new(KeyBuffer::PARITY_LEN);
-        let dec = *dec.correct(base64.as_slice(), None)?;
-        Ok(KeyBuffer::with_raw(array_ref![dec, 0, KeyBuffer::KEY_LEN]))
-    }
-
-    /// generate a key buffer from raw bytes (no correction)
-    pub fn with_raw(b: &[u8; KeyBuffer::KEY_LEN]) -> KeyBuffer {
-        KeyBuffer(*b)
-    }
-
-    /// generate a key buffer from raw bytes from two parts (no correction)
-    pub fn with_raw_parts(
-        a: &[u8; KeyBuffer::HALF_KEY_LEN],
-        b: &[u8; KeyBuffer::HALF_KEY_LEN],
-    ) -> KeyBuffer {
-        let mut buf: [u8; KeyBuffer::KEY_LEN] = [0; KeyBuffer::KEY_LEN];
-
-        buf[..KeyBuffer::HALF_KEY_LEN].clone_from_slice(&a[..KeyBuffer::HALF_KEY_LEN]);
-
-        buf[KeyBuffer::HALF_KEY_LEN..KeyBuffer::KEY_LEN]
-            .clone_from_slice(&b[..KeyBuffer::HALF_KEY_LEN]);
-
-        KeyBuffer(buf)
-    }
+use std::{convert::TryFrom, str};
 
-    /// render a base64url encoded user identity with reed-solomon parity bytes
-    pub fn render(&self) -> String {
-        let enc = Encoder::new(KeyBuffer::PARITY_LEN);
-        let enc = *enc.encode(&self.0);
-        base64::encode(&enc[..]).replace("+", "-").replace("/", "_")
-    }
+use hcid::*;
 
-    /// get the signature public key portion of this buffer
-    pub fn get_sig(&self) -> &[u8; KeyBuffer::HALF_KEY_LEN] {
-        array_ref![self.0, 0, KeyBuffer::HALF_KEY_LEN]
-    }
+pub type Base32 = String;
 
-    /// get the encryption public key portion of this buffer
-    pub fn get_enc(&self) -> &[u8; KeyBuffer::HALF_KEY_LEN] {
-        array_ref![self.0, KeyBuffer::HALF_KEY_LEN, KeyBuffer::HALF_KEY_LEN]
-    }
-}
-
-/// agent data that can be stored in the cas
-/// note thate the "address" of an agent entry is the base64url encoded
-/// public key identity string
+/// AgentId represents an agent in the Holochain framework.
+/// This data struct is meant be stored in the CAS and source-chain.
+/// Its key is the public signing key, and is also used as its address.
 #[derive(Serialize, Deserialize, Debug, Clone, PartialEq, DefaultJson)]
 pub struct AgentId {
     /// a nickname for referencing this agent
     pub nick: String,
-    /// the base64url encoded public identity string for this agent
-    pub key: String,
+    /// the encoded public signing key of this agent (the magnifier)
+    pub pub_sign_key: Base32,
```

Comment:
can we go ahead and add the `pub_enc_key` here too? or would you rather do that at a later time?

Reply:
Better to do that later, since enc doesn't work yet.
I'll add a TODO though.